gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/SchemaVersionLockFile.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.dist;
+
+import com.palantir.logsafe.Preconditions;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public final class SchemaVersionLockFile {
```

Comment:
Could we explore making this an Immutable as well (that holds a List<SchemaMigration>), and generate the file using Jackson serialization somehow? This will make it more reliable to consume for end clients, and will make the serialization itself not rely on manual string construction.

Reply:
Made it immutable. I don't think it's worth it to make it jackson serializable.